### PR TITLE
Issue #SC-2165 User progress details

### PR DESCRIPTION
**Pull request by pc0202**
Modified the update and get consumption for progressdetails

### CODE REVIEW

File: course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/ContentConsumptionActor.scala [hunk 1]
```diff
@@ -193,6 +197,11 @@ class ContentConsumptionActor @Inject() extends BaseEnrolmentActor {
         val inputStatus = inputContent.getOrDefault(JsonKey.STATUS, 0.asInstanceOf[AnyRef]).asInstanceOf[Number].intValue()
         val updatedContent = new java.util.HashMap[String, AnyRef]()
         updatedContent.putAll(inputContent)
+        jsonFields.foreach(field =>
+          if (inputContent.containsKey(field))
+            updatedContent.put(field, mapper.writeValueAsString(inputContent.get(field)))
+
+        )
```
Comment (maheshkumargangula) [inline course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/ContentConsumptionActor.scala:204]: Use `filter` to identify the Json keys from inputContent object and parse. After that, use putAll to set the map with parsed data.

File: course-mw/enrolment-actor/src/test/scala/org/sunbird/enrolments/CourseConsumptionActorTest.scala [hunk 2]
```diff
@@ -22,7 +23,94 @@ import scala.concurrent.duration.FiniteDuration
...
+                put("batchId", "0123")
+                put("contentId", "do_456")
+                put("progressdetails", "{}")
+                put("progressDetails", progressResponse)
+            }})
+        }})
+
+        ((requestContext: RequestContext, keyspace: _root_.scala.Predef.String, table: _root_.scala.Predef.String, filters: _root_.java.util.Map[_root_.scala.Predef.String, AnyRef], fields: _root_.java.util.List[_root_.scala.Predef.String]) => cassandraOperation.getRecords(requestContext, keyspace, table, filters, fields)).expects(*,*,*,*,*).returns(response)
+        val result = callActor(getStateReadRequestWithProgressField(), Props(new ContentConsumptionActor().setCassandraOperation(cassandraOperation, false)))
+        assert(null!= result)
```
Comment (maheshkumargangula) [inline course-mw/enrolment-actor/src/test/scala/org/sunbird/enrolments/CourseConsumptionActorTest.scala:48]: please add more assertion on the result object to check the progressDetails attribute value.

**Review by maheshkumargangula — CHANGES_REQUESTED** [review]

**Comment by maheshkumargangula** [comment]
@pritha-tarento - requested changes. Please check my comments and make the required changes.

File: course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/ContentConsumptionActor.scala (hunk 1 above)
Comment (pritha-tarento) [reply course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/ContentConsumptionActor.scala]: Resolved

File: course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/ContentConsumptionActor.scala [hunk 3]
```diff
@@ -293,12 +302,21 @@ class ContentConsumptionActor @Inject() extends BaseEnrolmentActor {
         val courseId = request.get(JsonKey.COURSE_ID).asInstanceOf[String]
         val contentIds = request.getRequest.getOrDefault(JsonKey.CONTENT_IDS, new java.util.ArrayList[String]()).asInstanceOf[java.util.List[String]]
         val fields = request.getRequest.getOrDefault(JsonKey.FIELDS, new java.util.ArrayList[String](){{ add(JsonKey.PROGRESS) }}).asInstanceOf[java.util.List[String]]
-        val contentsConsumed = getContentsConsumption(userId, courseId, contentIds, batchId, request.getRequestContext)
+        //default fields added ..
+        if (fields!=null && !fields.isEmpty ){
+            defaultFields = defaultFields ++ fields
+        }
```
[the remark this answers is not in the document]
Comment (pritha-tarento) [reply course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/ContentConsumptionActor.scala]: Resolved

File: course-mw/enrolment-actor/src/test/scala/org/sunbird/enrolments/CourseConsumptionActorTest.scala (hunk 2 above)
Comment (pritha-tarento) [reply course-mw/enrolment-actor/src/test/scala/org/sunbird/enrolments/CourseConsumptionActorTest.scala:52]: Resolved

File: course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/ContentConsumptionActor.scala (hunk 1 above)
Comment (pritha-tarento) [reply course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/ContentConsumptionActor.scala]: Added flatmap to build parse data.  Then used putAll to the map with parsed data.

**Comment by maheshkumargangula** [comment]
@pritha-tarento - the circleci build failed. Please check it.

**Comment by pritha-tarento** [comment]
> @pritha-tarento - the circleci build failed. Please check it.

@maheshkumargangula , i find the build failed message exists for the original repo (release-3.6.0).

**Pull request closed without merging**